packages/legend-studio-preset-query-builder/src/stores/QueryBuilderProjectionState.ts

```diff
@@ -555,4 +597,125 @@ export class QueryBuilderProjectionState {
       ),
     );
   }
+
+  *previewData(
+    node: QueryBuilderExplorerTreePropertyNodeData,
+  ): GeneratorFn<void> {
+    if (
+      !node.mapped ||
+      !this.queryBuilderState.querySetupState._class ||
+      !this.queryBuilderState.querySetupState.mapping
+    ) {
+      return;
+    }
+    if (
+      this.queryBuilderState.explorerState.previewDataState
+        .isGeneratingPreviewData
+    ) {
+      this.editorStore.applicationStore.notifyWarning(
+        `Can't preview data for property '${node.property.name}': another preview request is being executed`,
+      );
+      return;
+    }
+    this.queryBuilderState.explorerState.previewDataState.setPropertyName(
+      node.property.name,
+    );
+    this.queryBuilderState.explorerState.previewDataState.setIsGeneratingPreviewData(
+      true,
+    );
+    const propertyExpression = buildPropertyExpressionFromExplorerTreeNodeData(
+      this.queryBuilderState.explorerState.nonNullableTreeData,
+      node,
+      this.editorStore.graphState.graph,
+    );
+    const propertyType = node.property.genericType.value.rawType;
+    try {
+      switch (propertyType.path) {
+        case PRIMITIVE_TYPE.NUMBER:
+        case PRIMITIVE_TYPE.INTEGER:
+        case PRIMITIVE_TYPE.DECIMAL:
+        case PRIMITIVE_TYPE.FLOAT: {
+          const previewResult =
+            (yield this.editorStore.graphState.graphManager.executeMapping(
+              this.editorStore.graphState.graph,
+              this.queryBuilderState.querySetupState.mapping,
+              this.queryBuilderState.buildRawLambdaFromLambdaFunction(
+                buildNumericPreviewDataQuery(
```

Comment:
In preparation for the db builder, wouldn't it be better to have a more generic function we call that would do this switch case? something like `buildPreviewDataQuery` that calls the Numeric and NonNumericQuery builder.

Reply:
yeah but I was not coding against that yet, trying to get this out quick and (arguably not so) dirty. We will co-op when you do that part